core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java

```diff
@@ -793,35 +806,146 @@ public void testDeletingClosedIndexRemovesFiles() throws Exception {
                 .put(IndexMetaData.SETTING_SHADOW_REPLICAS, true)
                 .put(IndexMetaData.SETTING_SHARED_FILESYSTEM, true)
                 .build();
-        Settings idx2Settings = Settings.builder()
-                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 5)
-                .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
-                .put(IndexMetaData.SETTING_DATA_PATH, dataPath2.toAbsolutePath().toString())
-                .put(IndexMetaData.SETTING_SHADOW_REPLICAS, true)
-                .put(IndexMetaData.SETTING_SHARED_FILESYSTEM, true)
-                .build();
 
         prepareCreate(IDX).setSettings(idxSettings).addMapping("doc", "foo", "type=text").get();
-        prepareCreate(IDX2).setSettings(idx2Settings).addMapping("doc", "foo", "type=text").get();
-        ensureGreen(IDX, IDX2);
+        ensureGreen(IDX);
 
         int docCount = randomIntBetween(10, 100);
         List<IndexRequestBuilder> builders = new ArrayList<>();
         for (int i = 0; i < docCount; i++) {
             builders.add(client().prepareIndex(IDX, "doc", i + "").setSource("foo", "bar"));
-            builders.add(client().prepareIndex(IDX2, "doc", i + "").setSource("foo", "bar"));
         }
         indexRandom(true, true, true, builders);
-        flushAndRefresh(IDX, IDX2);
+        flushAndRefresh(IDX);
 
         logger.info("--> closing index {}", IDX);
         client().admin().indices().prepareClose(IDX).get();
 
-        logger.info("--> deleting non-closed index");
-        client().admin().indices().prepareDelete(IDX2).get();
-        assertPathHasBeenCleared(dataPath2);
         logger.info("--> deleting closed index");
         client().admin().indices().prepareDelete(IDX).get();
+
+        assertPathHasBeenCleared(dataPath);
+        assertIndicesDirsDeleted(nodes);
+    }
+
+    public void testDeletingIndexWithDedicatedMasterNodes() throws Exception {
```

Comment:
I don't think we need to get into this now. For now we can remove the test and just keep a test where we have some nodes with no shards assigned to them.